src/Sulu/Bundle/SnippetBundle/Content/SingleSnippetSelection.php

```diff
@@ -37,14 +37,21 @@ class SingleSnippetSelection extends SimpleContentType implements PreResolvableC
      */
     private $snippetReferenceStore;
 
+    /**
+     * @var ReferenceStoreInterface
+     */
+    private $snippetAreaReferenceStore;
+
     public function __construct(
         SnippetResolverInterface $snippetResolver,
         DefaultSnippetManagerInterface $defaultSnippetManager,
-        ReferenceStoreInterface $snippetReferenceStore
+        ReferenceStoreInterface $snippetReferenceStore,
+        ReferenceStoreInterface $snippetAreaReferenceStore
```

Comment:
We require here some bc handling:

```suggestion
        ReferenceStoreInterface $snippetReferenceStore,
        ?ReferenceStoreInterface $snippetAreaReferenceStore = null
```